### PR TITLE
Fix: linkpreview

### PR DESCRIPTION
**Pull request by wictornogueira**
# PR Details

This pr fixes link previews, which have been long broken.

## Related Issue

https://github.com/pedroslopez/whatsapp-web.js/issues/2294
https://github.com/pedroslopez/whatsapp-web.js/issues/2220

## How Has This Been Tested

Tested by sending a message containing a url 🤷

## Types of changes

- [ ] Dependency change
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly (index.d.ts).

### CODE REVIEW

**Review by opssemnik — APPROVED** [review]
Confirmed working using main branch lib 1.21.1-alpha.0
WWeb: 2.2332.15

File: src/util/Injected.js [hunk 1]
```diff
@@ -80,6 +80,14 @@ exports.ExposeStore = (moduleRaidStr) => {
         };
     }
 
+    // VERSION >= 2.2330.X
```
Comment (shirser121) [inline src/util/Injected.js:83]: Can you use `window.WWebJS.compareWwebVersions` to check if the version allowing that feature?

File: src/util/Injected.js [hunk 2]
```diff
@@ -207,6 +215,15 @@ exports.LoadUtils = () => {
...
                     options = { ...options, ...preview };
                 }
+            } else {
+                const link = window.Store.Validators.findLink(content);
+                if (link) {
+                    const preview = await window.Store.getLinkPreview(link).catch(() => null);
+                    if (preview) {
+                        preview.data.subtype = 'url';
+                        options = { ...options, ...preview.data };
+                    }
```
Comment (alechkos) [inline src/util/Injected.js:225]: I think you can't use `.catch()` with `await`
either `async`/`await` with `try...catch` block or `.then()` `.catch()`
```js
window.Store.getLinkPreview(link)
    .then(preview => {
        if (preview) {
            preview.data.subtype = 'url';
            options = { ...options, ...preview.data };
        }
    })
    .catch(() => null);
```

Comment (shirser121) [reply src/util/Injected.js:225]: Please use await with try and catch

Comment (alechkos) [reply src/util/Injected.js:225]:
```js
try {
    const preview = await window.Store.getLinkPreview(link)
    if (preview) {
        preview.data.subtype = 'url';
        options = { ...options, ...preview.data };
    }
} catch (error) { /** empty */ }
```

File: src/util/Injected.js (hunk 1 above)
Comment (wictornogueira) [reply src/util/Injected.js:83]: I am not 100% sure which version added that specific method. Besides, checking the method itself could be a safer play.

File: src/util/Injected.js (hunk 2 above)
Comment (wictornogueira) [reply src/util/Injected.js:225]: https://github.com/pedroslopez/whatsapp-web.js/pull/2378/commits/9a3900f9a53a48851b8b9b94c93c19c5b23d8bb1
Eslint was complaining about the empty catch block.

Comment (wictornogueira) [reply src/util/Injected.js:225]: +I don't plan on doing any fancy error handling, so, **functionally**, this is not different than a catch { preview = null }

Comment (alechkos) [reply src/util/Injected.js:225]: > [9a3900f](https://github.com/pedroslopez/whatsapp-web.js/commit/9a3900f9a53a48851b8b9b94c93c19c5b23d8bb1) Eslint was complaining about the empty catch block.

You can use
```js
catch { /** empty */ }
```

Comment (wictornogueira) [reply src/util/Injected.js:225]: I'm not big on the idea of tricking the lint for no reason, but I guess that's fair. Thank you @alechkos.

**Comment by vscheeren** [comment]
Confirmed working on WhatsApp Web Version 2.2333.11. Thank you

**Comment by shirser121** [comment]
Please resolve conflicts

**Review by hdem607 — APPROVED** [review]
Worked fine for me!
Thanks

**Review by hdem607 — APPROVED** [review]
Worked fine for me.
Clear to merge.
Thanks

**Comment by goldalworming** [comment]
how to update to this version, I npm update whatsapp-web.js@latest still not working

**Comment by hdem607** [comment]
> how to update to this version, I npm update whatsapp-web.js@latest still not working

This is not a merged fork. Is is not in *whatsapp-web.js* yet.
You can easy find on Google how to answer your own question.

**Comment by wictornogueira** [comment]
> oh sorry, I thought `merge branch 'main' into link-preview` means already merge

It means that `main` has been merged into `link-preview`, but not the other way around 😅

**Comment by nandobfer** [comment]
any ETA for this to be merged in main? Currently i have to choose between this running this pr or the alpha branch which fixes connection on new whatsapp web version

**Comment by goldalworming** [comment]
> any ETA for this to be merged in main? Currently i have to choose between this running this pr or the alpha branch which fixes connection on new whatsapp web version

I'm waiting it too

**Review by elhumbertoz — APPROVED** [review]
Excelent!

**Comment by renzhxr** [comment]
no work

**Review by jairocedenoadrian — APPROVED** [review]

**Comment by itaykrk** [comment]
Hi what's the status with the link preview fix? really looking forward for this one

**Comment by Todpig** [comment]
Someone send me an example of the code to set the link preview, I'm just passing { linkPreview: true } and it doesn't work

**Review by kennedysmartins — APPROVED** [review]
it is working correctly

**Pull request closed without merging**